pulp_rpm/app/downloaders.py

```diff
@@ -0,0 +1,49 @@
+from logging import getLogger
+from aiohttp.client_exceptions import ClientResponseError
+from pulpcore.plugin.download import HttpDownloader
+from urllib.parse import urljoin
+
+log = getLogger(__name__)
+
+
+class RpmDownloader(HttpDownloader):
+    """
+    Custom Downloader that automatically handles authentication token for SLES repositories.
+    """
+
+    def __init__(self, *args, **kwargs):
+        """
+        Initialize the downloader.
+        """
+        if 'sles_auth_token' in kwargs.keys():
+            self.sles_auth_token = kwargs.pop('sles_auth_token')
+        else:
+            self.sles_auth_token = None
+        super().__init__(*args, **kwargs)
+
+    async def _run(self, extra_data=None):
+        """
+        Download, validate, and compute digests on the `url`. This is a coroutine.
+
+        This method provides the same return object type and documented in
+        :meth:`~pulpcore.plugin.download.BaseDownloader._run`.
+
+        """
+        if self.sles_auth_token:
+            auth_param = '?' + self.sles_auth_token
```

Comment:
that's completely fine, just fwiw, f string can be used.
`f'?{self.sles_auth_token}'`

Reply:
I agree with fstring, to use features of python language